lkmpg.tex

```diff
@@ -1708,6 +1708,20 @@ \subsection{Spinlocks}
 Sometimes you may find comments in kernel source code stating that a function ``may sleep'', ``might sleep'', or more explicitly ``the caller should not hold a spinlock''.
 Those comments are hints that a function may implicitly sleep and must not be called in atomic contexts.
 
+Now, let's differentiate between a few types of spinlock functions in Linux kernel: \cpp|spin_lock()|, \cpp|spin_lock_irq()|, \cpp|spin_lock_irqsave()|, and \cpp|spin_lock_bh()|.
+
+\cpp|spin_lock()| does not allow the CPU to sleep while waiting for the lock, which makes it suitable for most use cases where the critical section is short.
+However, this is problematic for RT Linux because spinlocks in this configuration behave as sleeping locks.
```

Comment:
Don't use the term "RT Linux" which might be confusing with [RTLinux](https://en.wikipedia.org/wiki/RTLinux). Instead, use [real-time Linux](https://ubuntu.com/blog/what-is-real-time-linux-i), the fully qualified form.